### PR TITLE
[chore] observation_table: combine stats into single query

### PR DESCRIPTION
**Pull request by jevonyeoh**
## Description
Replace the `point_in_time_stats` query to utilize the min and max stats query from the describe function.

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/service/observation_table.py [hunk 1]
```diff
@@ -295,12 +273,12 @@ async def validate_materialized_table_and_get_metadata(
...
         validate_columns_info(columns_info, skip_entity_validation_checks)
         # Get point in time metadata
-        point_in_time_stats = await get_point_in_time_stats(
-            db_session=db_session,
-            destination=table_details,
-        )
+        # point_in_time_stats = await get_point_in_time_stats(
+        #     db_session=db_session,
+        #     destination=table_details,
+        # )
```
Comment (gee-senbong) [inline featurebyte/service/observation_table.py:279]: Is this intended?

**Review by gee-senbong — APPROVED** [review]

File: featurebyte/service/observation_table.py [hunk 2]
```diff
@@ -256,7 +224,17 @@ async def _get_column_name_to_entity_count(
...
                 "unique", col_name
             ]
-        return column_name_to_count
+        least_recent_time_str = describe_stats_dataframe.loc[  # pylint: disable=no-member
+            "min", "POINT_IN_TIME"
+        ]
+        least_recent_time_str = _convert_ts_to_str(least_recent_time_str)
+        most_recent_time_str = describe_stats_dataframe.loc[  # pylint: disable=no-member
+            "max", "POINT_IN_TIME"
+        ]
```
Comment (gee-senbong) [inline featurebyte/service/observation_table.py:233]: Should we use `SpecialColumnName.POINT_IN_TIME` here?

File: featurebyte/service/observation_table.py [hunk 3]
```diff
@@ -48,43 +47,12 @@ class PointInTimeStats:
...
-        if current_timestamp.tzinfo is not None:
-            current_timestamp = current_timestamp.tz_convert("UTC").tz_localize(None)
-        current_timestamp = current_timestamp.isoformat()
-        return cast(str, current_timestamp)
-
-    most_recent_point_in_time_str = _convert_ts_to_str(most_recent_point_in_time)
-    earliest_str = _convert_ts_to_str(least_recent_point_in_time)
-    return PointInTimeStats(most_recent=most_recent_point_in_time_str, least_recent=earliest_str)
+def _convert_ts_to_str(timestamp_str: str) -> str:
+    current_timestamp = pd.Timestamp(timestamp_str)
```
Comment (YS-L) [inline featurebyte/service/observation_table.py:51]: Not a new issue, but this `current_timestamp` variable name is a bit misleading - I thought it refers to the current time. Might want to change it to something like just `timestamp` or `timestamp_obj`.

Comment (jevonyeoh) [reply featurebyte/service/observation_table.py:51]: will rename!

File: featurebyte/service/observation_table.py (hunk 1 above)
Comment (jevonyeoh) [reply featurebyte/service/observation_table.py:279]: oops forgot to delete!

**Review by YS-L — APPROVED** [review]

**Pull request merged**